src/tests/dbus-tests/test_90_swapspace.py

```diff
@@ -0,0 +1,86 @@
+
+    def get_device(self, dev_name):
+        dev = self.get_object('', '/block_devices/' + os.path.basename(dev_name))
+        return dev
+
+
+    def get_swapspace_iface(self, device):
+        iface = dbus.Interface(device, dbus_interface=self.iface_prefix + '.Swapspace')
+        return iface
+
+
+    def setUp(self):
+        # create new fake swap device
+        subprocess.call(['modprobe', 'scsi_debug'])
+        dirs = glob.glob('/sys/bus/pseudo/drivers/scsi_debug/adapter*/host*/target*/*:*/block')
+
+        with open('/sys/bus/pseudo/drivers/scsi_debug/ptype', 'w') as f:
+            f.write('0')  # '0' == created devices will be disks
+        with open('/sys/bus/pseudo/drivers/scsi_debug/num_tgts', 'w') as f:
+            f.write('1')  # change desired number of new targets to 1
+        with open('/sys/bus/pseudo/drivers/scsi_debug/add_host', 'w') as f:
+            f.write('1')  # generate a new device (add_host also applies changes)
+        subprocess.call(['udevadm', 'settle'])
+
+        new_dirs = []
+        # rescan directories; we cannot tell which file is the new one...
+        while len(new_dirs) < len(dirs) + 1:
+            new_dirs = glob.glob('/sys/bus/pseudo/drivers/scsi_debug/adapter*/host*/target*/*:*/block')
+            time.sleep(0.1)
+
+        # ...so we just diff all devices with the previous scan result to get the new one
+        new_dirs = list(set(new_dirs) - set(dirs))
+        assert len(new_dirs) == 1
+
+        self.dev = os.listdir(new_dirs[0])
+        assert len(self.dev) == 1
+        self.dev = '/dev/' + self.dev[0]
+        assert os.path.exists(self.dev)
+
+        print('\n')
+        subprocess.call(['mkswap', self.dev])
+        subprocess.call(['udevadm', 'settle'])
+        self.device = self.get_device(self.dev)
+        self.iface = self.get_swapspace_iface(self.device)
+
+
+    def tearDown(self):
+        device = self.dev.split('/')[-1]
+        if os.path.exists('/sys/block/' + device):
+            with open('/sys/block/%s/device/delete' % device, 'w') as f:
+                f.write('1')
+            while os.path.exists(device):
+                time.sleep(0.1)
+            subprocess.call(['udevadm', 'settle'])
+            subprocess.call(['modprobe', '-r', 'scsi_debug'])
+
+
+    def test_10_start_stop(self):
+        # Active property is now in undefined state
+
+        self.iface.Start(self.no_options)
+        active = self.get_property(self.iface, '.Swapspace', 'Active')
+        self.assertEqual(active, True) # swap should be active
+
+        self.iface.Stop(self.no_options)
+        active = self.get_property(self.iface, '.Swapspace', 'Active')
+        self.assertNotEqual(active, True) # swap shouldn't be active
+
+        self.iface.Start(self.no_options)
+        active = self.get_property(self.iface, '.Swapspace', 'Active')
+        self.assertEqual(active, True) # swap should be active
+
+        # cleanup
+        self.iface.Stop(self.no_options)
```

Comment:
Cleanup should be part of `tearDown` (or you can use `addCleanup`) -- if one of the assertion fails the stop method won't be executed. It also might be better to call `swapoff` in cleanup.